runtime/lua/vim/lsp/buf.lua

```diff
@@ -87,6 +95,33 @@ function M.hover(config)
       else
         vim.list_extend(contents, util.convert_input_to_markdown_lines(result.contents))
       end
+      local range = result.range
+      if range and config.apply_highlight then
```

Comment:
Hmmm given that we now support multiple responses for a single hover request (cc @lewis6991 @mfussenegger), this could highlight several ranges.

What should we do if a client doesn't include a highlight range but another one does? What if the ranges are different?

Reply:
Yeah I was wondering about that... right now all provided ranges will be highlighted, so they will appear combined if they overlap at all. This seemed like the most obvious solution but maybe there are cases I didn't consider where this would be unintuitive?

Reply:
If we end up using cycling for handling multiple hover responses (see https://github.com/neovim/neovim/pull/31029#issuecomment-2453280867 for more info on the idea) then we could potentially match the highlighted range with the current hover. I don't think there's an issue with the current approach of "over-highlighting" but extmarks aren't my area of expertise.